test/unit/datetime.result

```diff
@@ -1,4 +1,4 @@
-1..4
+1..5
```

Comment:
Unit tests produce a valid TAP13 output, and result files are not needed for when you are using `assert()` or similar checks in tests. Could we drop result file at least for datetime test?

Reply:
Good question! You know it bothers me as well, but by some reason test-run expected here result file. Do you know what should I do to make it accept our TAP output without redundant result file?

Reply:
Apply patch below and remove `datetime.result` file.

```diff
commit fec49cc2d87cf8a1337258ab239360e383f513f6
Author: Sergey Bronnikov <sergeyb@tarantool.org>
Date:   Mon Jun 8 13:43:08 2020 +0000

    test: update unit test lib to produce TAP-compliant output
    
    TAP (Test Anything Protocol) output must indicate that
    this is TAP13 in the first line. See specification [1].
    
    [1] https://testanything.org/tap-version-13-specification.html
    
    Part of #5000

diff --git a/test/unit/unit.c b/test/unit/unit.c
index d12b66f91..302dce2ef 100644
--- a/test/unit/unit.c
+++ b/test/unit/unit.c
@@ -26,6 +26,9 @@ plan(int count)
 	tests_done[level] = 0;
 	tests_failed[level] = 0;
 
+	if (level == 0)
+		printf("TAP version 13\n");
+
 	_space(stdout);
 	printf("%d..%d\n", 1, plan_test[level]);
 }
diff --git a/test/unit/unit.h b/test/unit/unit.h
index 43c301b28..e207255fe 100644
--- a/test/unit/unit.h
+++ b/test/unit/unit.h
@@ -33,9 +33,6 @@
 #include <stdio.h>
 #include <stdlib.h> /* exit() */
 
-#define header() printf("\t*** %s ***\n", __func__)
-#define footer() printf("\t*** %s: done ***\n", __func__)
-
 #define fail(expr, result) do {					\
 	fprintf(stderr, "Test failed: %s is %s at %s:%d, in function '%s'\n",\
 		expr, result, __FILE__, __LINE__, __func__);		\
@@ -129,6 +126,16 @@ int check_plan(void);
 	}						\
 }
 
+#define header() do {					\
+	_space(stdout);					\
+	printf("# *** %s ***\n", __func__);		\
+} while(0)
+
+#define footer() do {					\
+	_space(stdout);					\
+	printf("# *** %s: done ***\n", __func__);	\
+} while(0)
+
 #if defined(__cplusplus)
 }
 #endif /* defined(__cplusplus) */
```

Reply:
Also put these commits on top of your branch:

- https://github.com/tsafin/tarantool/commit/fec49cc2d87cf8a1337258ab239360e383f513f6
- https://github.com/tsafin/tarantool/commit/9296f8f501a7c1b6cb713849e0e3b9ed85443660

Reply:
Well, after struggle with all consequences of this modified `unit.h` I've taken slightly different approach. Now I introduced new `tapunit.h` which is actually amalgamation of former `unit.c`/`unit.h` (as it used to be ages ago, before Dima Oboukhov has separated them).
Also tapunit.h` incorporated all changes to `unit.h` which were necessary to make output TAP compatible. The trick now is to use that tapunit.h in 1 by 1 case for each test we want to remove result file. And we do want to get rid of it for datetime test.

* https://github.com/tsafin/tarantool/commit/4e256617e3a5f86b786e6f592a6ba9649cd2d7c3
* https://github.com/tsafin/tarantool/commit/4c8c7d49157dafa767c1ab33bcf0806437ca0985

Reply:
<details>
<summary>Off-topic</summary>

And now we'll go into a war regarding static variables in a header file? Will discuss what should go first: header or plan? Will involve more interested developers (me and Vlad as least) into the discussion right here?

I think that such twists are not productive. If it would be obvious small change, which has no points to discuss, it would be okay. But this patch is almost exactly what I created and not sent in early 2020, because it looks dubious for me. There are things to think around.

Why not do it outside of the datetime PR? Pros: we'll see the diff in `unit.[ch]`, we'll see all required changes in tests, we can discuss with its own timeline, its own reviewers list and without mixing irrelevant comments into one PR.

(Feel free to do anything here, but don't be surprised by increased time of review.)
</details>